### PR TITLE
Added new built-in Policies into the Initiative

### PR DESCRIPTION
**Pull request by pkorolo**
## Overview/Summary

This PR brings additional existing built-in DINE Policies into this initiative, as per performed gap analysis

## This PR fixes/adds/changes/removes

1. Adds the majority of the existing DINE Policies, that were not initially added in the Initiative
2. The "Storage Table" Policies were **not** included, due to conflict with the "Cosmos DB" (Table API) Policy, which was included
3. The IoT Central & Bot Service Policies were **not** included, due to limited or conflicting information in the documentation, regarding the respective required DNS Zone integration

### Breaking Changes

1. Private DNS Zone for "Azure Migrate" ("privatelink.prod.migration.windowsazure.com"), is **not** included in the RI, hence either the zone must be **added** as well, or the respective Policy for "Azure Migrate" must be **removed**

### Testing URLs

The below URLs can be updated where the placeholders are, look for `{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}` & `{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}`, to allow you to test your portal deployment experience.

> Please also replace the curly brackets on the placeholders `{}`

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fpkorolo%2FEnterprise-Scale%2FUpdate-Custom-PrivateDNSInitiative%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fpkorolo%2FEnterprise-Scale%2FUpdate-Custom-PrivateDNSInitiative%2FeslzArm%2Feslz-portal.json)

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Review by jtracey93 — DISMISSED** [review]
Nice stuff @pkorolo. 

A couple of things:

1. Can we update the Whats New Wiki page? - https://github.com/Azure/Enterprise-Scale/blob/main/docs/wiki/Whats-new.md
2. Do we also need to update the assignments?
    - Public - https://github.com/Azure/Enterprise-Scale/blob/main/eslzArm/managementGroupTemplates/policyAssignments/DINE-PrivateDNSZonesPolicyAssignment.json
    - Fairfax - https://github.com/Azure/Enterprise-Scale/blob/main/eslzArm/managementGroupTemplates/policyAssignments/gov/fairfaxDINE-PrivateDNSZonesPolicyAssignment.json

As for the Azure Migrate piece I think its worth adding this so up to you if you want to wrap into this branch/PR - might be easiest?

Might be worth you also picking up #1073 as part of this.

Let me know your thoughts

Thanks

Jack

**Comment by pkorolo** [comment]
Hello @jtracey93 

To your points:

I am in very limited availability as we speak, due to my "WAF-Resiliency" assignment. But I will have a clearer picture early next week, if I can follow up with additional tasks or not; but:

1. if I cannot take this (documentation - "What's new"), I can help the one that will
2. Assignments **_definitely_** need to be updated; again, not sure if I can take that next week - btw, I think (maybe) I should have put **default parameters** in the Initiative itself for **all** the "groupIDs", so that the one who will eventually work with this, will have to _only_ supply the "ZoneIDs" (as in the previous iteration), what do you think? 
[If "yes", I _can_ work with this (minor) addition and submit a new PR (?) ]

Last, for the deployment of Azure Migrate Private DNS Zone itself, if I would take that eventually, where should I touch this?

Thank you.

**Comment by pkorolo** [comment]
@jtracey93 can you check this conflict please? 

Btw, I have updated 

- the Policy Definition with default values in GroupIDs
- also updated the Azure Public assignments, and 
- added the extra private DNS zone for Azure migrate

please review, thank you!

File: src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json
Comment (pkorolo) [reply src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json:37]: @krowlandson , 

Thank you for your help resolving the conflict in my PR.
Now on your question, I might have misunderstood what you are asking, but some of those built-in, like the one in the example ( /providers/Microsoft.Authorization/policyDefinitions/6dd01e4f-1be1-4e80-9d0b-d109e04cb064 ), require those parameters when deployed

Screenshot 2022-11-16 171744

Screenshot 2022-11-16 171936

Comment (jtracey93) [reply src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json:37]: ah okay as they dont have defaults. @krowlandson happy with this otherwise and now this is cleared up?

**Comment by jtracey93** [comment]
@pkorolo is this ready for a re-review? If so, i dont see the Whats New page being updated.

My plan is to merge #1085 first before this, so may be worth you holding off to pull in the latest from main again once this is merged 🙂 (should merge this tonight/tomorrow)

File: src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json
Comment (krowlandson) [reply src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json:37]: If I understand the policy construct correctly, I believe the above "feature" of these policy definitions is due to a given resource type having multiple `groupId` values for different sub-resources. What I believe they have done is created a single generic policy definition which needs to be assigned multiple times, once for each sub-resource type (and therefore `groupId`) associated with the main resource type.

Within this initiative, you are declaring the same policy definition multiple times, once for each sub-resource type. As such, I believe you don't need to elevate the `groupId` value to a parameter within the initiative, but instead should set the value as hardcoded within the `policyDefinitions` section.

Happy to discuss offline if this still isn't clear.

Comment (pkorolo) [reply src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json:37]: @krowlandson perfectly clear! if also @jtracey93 shares the same view, I can update the PR accordingly.

Please let me know!

**Comment by pkorolo** [comment]
@jtracey93 , code-wise this is ready to be reviewed.

Based of course on @krowlandson comments above, there is a possibility that the main initiative definition would be updated again; please let me know accordingly.

As for the  What's New page, I was planning to do this last. Again, let me know when I can proceed.

**Comment by pkorolo** [comment]
Hello @krowlandson @jtracey93 I have put the latest changes there, as per previous recommendations about the "groupID" (default) values, and also updated the "what's new' respectively.
I think it is ready for review. Thank you!

**Review by krowlandson — APPROVED** [review]
LGTM

File: eslzArm/managementGroupTemplates/policyAssignments/DINE-PrivateDNSZonesPolicyAssignment.json
Comment (krowlandson) [inline eslzArm/managementGroupTemplates/policyAssignments/DINE-PrivateDNSZonesPolicyAssignment.json:67]: Just realised that this has probably been wrong since creation, as this is the namespace for Azure SignalR WebPubSub, but the parameter name (in my opinion) leads me to believe this was intended for Azure Web Apps.

The policy mapped to this parameter is indeed Azure SignalR WebPubSub so it technically maps correctly, but this doesn't feel correct.

File: eslzArm/managementGroupTemplates/policyAssignments/DINE-PrivateDNSZonesPolicyAssignment.json
Comment (krowlandson) [inline eslzArm/managementGroupTemplates/policyAssignments/DINE-PrivateDNSZonesPolicyAssignment.json:69]: Similarly, this appears to be the namespace for App Configuration Stores - making this an equally ambiguous parameter name. Would probably be a breaking change to reverse some of this though unfortunately.

File: eslzArm/managementGroupTemplates/policyAssignments/DINE-PrivateDNSZonesPolicyAssignment.json
Comment (krowlandson) [inline eslzArm/managementGroupTemplates/policyAssignments/DINE-PrivateDNSZonesPolicyAssignment.json:71]: And another where the name is misleading given the presence of IoT hubs, although this one may be easier to switch around if needed.

**Review by krowlandson — DISMISSED** [review]
LGTM 🔥 Thank you

**Review by krowlandson — DISMISSED** [review]
LGTM

**Review by krowlandson — APPROVED** [review]
LGTM

**Comment by pkorolo** [comment]
Branch to be deleted

**Pull request merged**
